keycloak/extensions/src/main/java/com/github/thomasdarimont/keycloak/custom/endpoints/filter/ProtectedResourceProviderFactory.java

```diff
@@ -0,0 +1,37 @@
+package com.github.thomasdarimont.keycloak.custom.endpoints.filter;
+
+import com.google.auto.service.AutoService;
+import org.keycloak.Config;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.services.resource.RealmResourceProvider;
+import org.keycloak.services.resource.RealmResourceProviderFactory;
+
+
+@AutoService(RealmResourceProviderFactory.class)
+public class ProtectedResourceProviderFactory implements RealmResourceProviderFactory {
```

Comment:
I suggest to remove this class and add the `ProtectedResource` example as a SubResource of the `CustomResource`.

See: 
https://github.com/thomasdarimont/keycloak-project-example/blob/main/keycloak/extensions/src/main/java/com/github/thomasdarimont/keycloak/custom/endpoints/CustomResource.java#L86